eclipse/src/saros/editor/internal/ContributionAnnotationManager.java

```diff
@@ -122,73 +108,72 @@ public ContributionAnnotationManager(
    */
   public void insertAnnotation(IAnnotationModel model, int offset, int length, User source) {
 
-    if (!contribtionAnnotationsEnabled) return;
-
-    if (length < 0) // why is 0 len allowed ?
-    return;
+    if (!contribtionAnnotationsEnabled || length < 0) return;
```

Comment:
insertAnnotation is called by EditorManager#replaceText, which is also called if a deletion of a char is performed. This leads to length == 0.I would like to change the handling, but I had to stop with refactoring and rework because the PR would be too large.

I restructured the code in order to make the cases more clear.